### PR TITLE
feat(legacy-preset-chart-deckgl): Add ,.1f and ,.2f value formats to deckgl charts

### PR DESCRIPTION
**Pull request by kgabryje**
### SUMMARY
This PR adds 2 new formattings to Legend Format control in deckgl charts.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
image

### TESTING INSTRUCTIONS
Verify that ,.1f and ,.2f legend formattings are available and work correctly

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

https://app.shortcut.com/preset/story/37656/add-1f-and-2f-to-the-legend-format-option-for-deck-gl-charts

### CODE REVIEW

**Review by michael-s-molina — APPROVED** [review]
LGTM

File: superset-frontend/plugins/legacy-preset-chart-deckgl/src/utilities/controls.jsx [hunk 1]
```diff
@@ -37,6 +37,8 @@ export const D3_FORMAT_OPTIONS = [
   [',.1%', ',.1% (12345.432 => 1,234,543.2%)'],
   ['.3%', '.3% (12345.432 => 1234543.200%)'],
   ['.4r', '.4r (12345.432 => 12350)'],
+  [',.1f', ',.1f (12345.432 => 12,345.4)'],
```
Comment (zhaoyongjie) [inline superset-frontend/plugins/legacy-preset-chart-deckgl/src/utilities/controls.jsx:40]: ﻿﻿﻿I remember that users can enter d3format manually. Why do we need to create a format in deckgl that is not consistent with all other charts?

image

**Review by zhaoyongjie — CHANGES_REQUESTED** [review]
In order to provide more format for this chart,  we just add `freeForm: true` for format control in Deckgl.

**Comment by kgabryje** [comment]
Thanks for pointing that out @zhaoyongjie. @yousoph @rusackas should we add those formatters in other charts as well? Should we quick and simple.

**Comment by yousoph** [comment]
Thanks for pointing that out, @zhaoyongjie ! 
@kgabryje I think it makes sense to add the freeform formatter to deck.gl (and any other chart types it's missing from)

**Comment by kgabryje** [comment]
@zhaoyongjie @yousoph Added `,.1f` and `,.2f` to `D3_FORMAT_OPTIONS` globally, added `freeForm` to deckgl, also removed duplicated code in deckgl plugins.

**Review by zhaoyongjie — APPROVED** [review]
Thanks for the improvement!

**Pull request merged**